tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,398 @@
+from pathlib import Path
+from unittest import skipIf
+
+import boto3
+from botocore.exceptions import ClientError
+import docker
+from botocore.config import Config
+from parameterized import parameterized
+
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from samcli.lib.bootstrap.companion_stack.data_types import CompanionStack
+from samcli.lib.config.samconfig import DEFAULT_CONFIG_FILE_NAME
+from samcli.lib.utils.resources import (
+    AWS_APIGATEWAY_RESTAPI,
+    # AWS_APIGATEWAY_V2_API,
+    AWS_LAMBDA_FUNCTION,
+    AWS_STEPFUNCTIONS_STATEMACHINE,
+)
+from tests.integration.buildcmd.build_integ_base import BuildIntegBase
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_SLEEP = 3
+API_SLEEP = 5
+SFN_SLEEP = 10
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+
+LOG = logging.getLogger(__name__)
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+class TestSync(BuildIntegBase, SyncIntegBase, PackageIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.docker_client = docker.from_env()
+        cls.local_images = [
+            ("public.ecr.aws/sam/emulation-python3.7", "latest"),
+        ]
+        # setup some images locally by pulling them.
+        for repo, tag in cls.local_images:
+            cls.docker_client.api.pull(repository=repo, tag=tag)
+            cls.docker_client.api.tag(f"{repo}:{tag}", "emulation-python3.7", tag="latest")
+            cls.docker_client.api.tag(f"{repo}:{tag}", "emulation-python3.7-2", tag="latest")
+
+        # setup signing profile arn & name
+        cls.signing_profile_name = os.environ.get("AWS_SIGNING_PROFILE_NAME")
+        cls.signing_profile_version_arn = os.environ.get("AWS_SIGNING_PROFILE_VERSION_ARN")
+
+        PackageIntegBase.setUpClass()
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "sync")
+
+        # To be used for code sync integ tests
+        # template_path = str(cls.test_data_path.joinpath("template.yaml"))
+        # cls.shared_stack_name = "accelerate-shared-stack"
+        # LOG.info(cls.shared_stack_name)
+
+        # command = "samdev" if os.getenv("SAM_CLI_DEV") else "sam"
+        # sync_command_list = [command, "sync", "-t", template_path, "--dependency-layer", "--stack-name", cls.shared_stack_name]
+        # run_command_with_input(sync_command_list, "y\n".encode())
```

Comment:
Do we need these commented-out lines?

Reply:
This is useful for the next task so I kept it here

Reply:
Wouldn't code sync be in a different test case file?